### PR TITLE
feat(price): add validation for max active prices

### PR DESCRIPTION
**Pull request by omkar273**
> [!IMPORTANT]
> Enforces a maximum of 1000 active prices per entity in price creation functions, with tests verifying this limit and handling validation skipping.
> 
>   - **Behavior**:
>     - Enforces a maximum of 1000 active prices per entity in `CreatePrice` and `CreateBulkPrice` in `price.go`.
>     - Introduces `MAX_ACTIVE_PRICES` constant in `model.go` to define the limit.
>     - Skips validation if `SkipEntityValidation` is true.
>   - **Tests**:
>     - Adds tests in `price_test.go` to verify price limit enforcement, including edge cases and validation skipping.
>     - Tests scenarios like exceeding the limit, exact limit, and mixed validation skipping.
>   - **Misc**:
>     - Updates `priceFilterFn` in `inmemory_price_store.go` to include entity type filtering.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=flexprice%2Fflexprice&utm_source=github&utm_medium=referral)<sup> for 984e15229ae0390c80ceeb9b1ddb042d74d2dc94. You can [customize](https://app.ellipsis.dev/flexprice/settings/summaries) this summary. It will automatically update as commits are pushed.</sup>

## Summary by CodeRabbit

* **New Features**
  * Enforced maximum of 1,000 active prices per entity for single and bulk price creation; attempts that exceed the limit now return validation errors.
  * Bulk creation supports per-item validation bypass to skip entity-limit checks.

* **Behavior Changes**
  * Listing/filtering now respects an entity-type filter when provided.

* **Tests**
  * Added comprehensive tests covering bulk creation limits, mixed scenarios, and validation bypass.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: internal/service/price.go [hunk 1]
```diff
@@ -60,6 +60,26 @@ func (s *priceService) CreatePrice(ctx context.Context, req dto.CreatePriceReque
 		if err := s.validateEntityExists(ctx, req.EntityType, req.EntityID); err != nil {
 			return nil, err
 		}
+
+		// Validate that the entity has less than 1000 active prices
+		filter := types.NewNoLimitPriceFilter().
+			WithEntityIDs([]string{req.EntityID}).
+			WithEntityType(req.EntityType)
```
Comment (nkmishra1997) [inline internal/service/price.go:68]: should have published and is_active check

File: internal/service/price.go [hunk 2]
```diff
@@ -224,6 +244,57 @@ func (s *priceService) CreateBulkPrice(ctx context.Context, req dto.CreateBulkPr
...
+		// Check if we should skip validation (check first item with this entityID)
+		itemsForEntity := entityGroups[entityID]
+	
+		firstItem := itemsForEntity[0]
+		if firstItem.SkipEntityValidation {
+			continue
+		}
+
+		// Count existing active prices for this entity
+		filter := types.NewNoLimitPriceFilter().
```
Comment (nkmishra1997) [inline internal/service/price.go:268]: move this in a helper method
